### PR TITLE
Support streaming streaming responses for callable functions.

### PR DESCRIPTION
**Pull request by taeold**
The new `.stream()` API allows the client to consume streaming responses from the WIP streaming callable functions in [Firebase Functions Node.js SDK](https://github.com/firebase/firebase-functions).

When client makes a request to the callable function w/ header `Accept: text/event-stream`, the callable function responds with response chunks in [Server-Sent Event](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events) format.

The sdk changes here abstracts over the wire-protocol by parsing the response chunks and returning an instance of a `AsyncIterable` to consume to data:

```ts
import { getFunctions, httpsCallable } from "firebase/functions";

const functions = getFunctions();
const generateText = httpsCallable(functions, 'generateText');
const resp = await generateText.stream(
  { text: 'What is your favorite Firebase service and why?' },
  { signal: AbortSignal.timeout(60_000) },
);
try {
  for await (const message of resp.stream) {
     console.log(message); // prints "foo", "bar"
  }
  console.log(await resp.data) // prints "foo bar"
} catch (e) {
  // FirebaseError(code='cancelled', message='Request was cancelled.');
  console.error(e)
}
```

### CODE REVIEW

File: packages/functions/src/service.ts
Comment (pavelgj) [inline packages/functions/src/service.ts:490]: this logic doesn't work if the server happens to return multiple data lines in a single chunk

Comment (taeold) [reply packages/functions/src/service.ts:533]: fixed.

**Review by hsubox76 — COMMENTED** [review]
CI stuff that you want to do before you finalize and merge:

(1) To address the formatting CI failure, run `yarn format` locally.
(2) To address the docgen failure, run `yarn docgen:all` locally. This will modify some files in `docs-devsite` which will require a techwriter review approval in addition to a JS core team review. go/firebase-contacts tells me the techwriter assigned to CF3 is Eric Gilmore.

File: packages/functions-types/index.d.ts
Comment (taeold) [reply packages/functions-types/index.d.ts:1]: Got it!

File: packages/functions/package.json
Comment (taeold) [reply packages/functions/package.json:41]: whoops i think this is an artifact of a merge gone wrong. Reverted.

Comment (taeold) [reply packages/functions/package.json:40]: On my dev machine, only 127.0.0.1 works (due to changes in recent Mac version that resolves localhost to ipv6), but I can revert the change if preferred.

File: packages/functions/src/callable.test.ts
Comment (taeold) [reply packages/functions/src/callable.test.ts:109]: hm I'm not even sure what I was trying to achieve with this change.

Reverting instead.

File: packages/functions/src/public-types.ts
Comment (taeold) [reply packages/functions/src/public-types.ts:84]: You are right that it would depend on the server imlementation.

We do try to provide the right primitives for developers to detect client dropping off, but who knows whether the developer would properly handle client connection drops.

See related server SDK PR: https://github.com/firebase/firebase-functions/pull/1635.

**Review by hsubox76 — APPROVED** [review]

File: packages/functions/src/public-types.ts
Comment (egilmorez) [inline packages/functions/src/public-types.ts:78]: about how stream call >>> about how a stream call OR about how stream calls

File: packages/functions/src/public-types.ts
Comment (egilmorez) [inline packages/functions/src/public-types.ts:83]: Suggest backticks for AbortSignal and caps for HTTP.

File: packages/functions/src/public-types.ts
Comment (egilmorez) [inline packages/functions/src/public-types.ts:88]: uses limited-use App Check token >>> uses a limited-use App Check token OR uses limited-use App Check tokens

File: docs-devsite/functions.httpscallable.md
Comment (egilmorez) [inline docs-devsite/functions.httpscallable.md:12]: I'm puzzled about why this is new/green/add when it seems to have existed already in line 43 of public_types.ts, iiuc.  

In any case, it would be great to have clarity on whether we really mean "Google Cloud Functions" or whether we should stick with just "Cloud Functions" or even "Cloud Functions for Firebase."

**Review by egilmorez — COMMENTED** [review]
A few thoughts and tweaks, thanks!

File: docs-devsite/functions.httpscallable.md
Comment (taeold) [reply docs-devsite/functions.httpscallable.md:12]: It's been a while since we made significant changes to this code, and it's possible the these type definition preceeds the devsite doc generation process I'm running here.

Comment (taeold) [reply docs-devsite/functions.httpscallable.md:12]: I'll stick with Cloud Functions?

Comment (egilmorez) [reply docs-devsite/functions.httpscallable.md:12]: SG, thanks!

**Review by egilmorez — APPROVED** [review]

**Pull request merged**
